packages/react-components/src/components/VideoGallery/useVideoStreamLifecycleMaintainer.ts

```diff
@@ -6,22 +6,45 @@ import { VideoStreamOptions, CreateVideoStreamViewResult, ScalingMode } from '..
 import { Cancellable, useCancellableTask } from '../utils/useCancellableTask';
 
 /**
- * Helper hook to maintain the video stream lifecycle. This calls onCreateStreamView and onDisposeStreamView
- * appropriately based on react lifecycle events and prop changes.
- * This also handles calls to view.update* appropriately such as view.updateScalingMode().
- *
+ * Props needed for {@link useVideoStreamLifecycleMaintainer}
  * @private
  */
-export const useVideoStreamLifecycleMaintainer = (props: {
+export interface VideoStreamLifecycleMaintainerProps {
   isStreamAvailable?: boolean;
   renderElementExists?: boolean;
   isMirrored?: boolean;
   scalingMode?: ScalingMode;
-  onCreateStreamView?: (options?: VideoStreamOptions) => Promise<void | CreateVideoStreamViewResult>;
-  onDisposeStreamView?: () => void;
-}): void => {
-  const { onCreateStreamView, onDisposeStreamView, isStreamAvailable, renderElementExists, isMirrored, scalingMode } =
-    props;
+  remoteParticipantId?: string;
+  onCreateLocalStreamView?: (options?: VideoStreamOptions) => Promise<void | CreateVideoStreamViewResult>;
+  onDisposeLocalStreamView?: () => void;
+  onCreateRemoteStreamView?: (
+    userId: string,
+    options?: VideoStreamOptions
+  ) => Promise<void | CreateVideoStreamViewResult>;
+  onDisposeRemoteStreamView?: (userId: string, options?: VideoStreamOptions) => Promise<void>;
+  isScreenSharingOn?: boolean;
+}
+
+/**
+ * Helper hook to maintain the video stream lifecycle. This calls onCreateStreamView and onDisposeStreamView
+ * appropriately based on react lifecycle events and prop changes.
+ * This also handles calls to view.update* appropriately such as view.updateScalingMode().
+ *
+ * @private
+ */
+export const useVideoStreamLifecycleMaintainer = (props: VideoStreamLifecycleMaintainerProps): void => {
+  const {
```

Comment:
tangent: I don't favor this practice of destructuring large `props` structs. 

- It just adds lines of code that all readers ignore. The moment we ignore something while we read code, there's an opportunity for bugs to lurk.
- I think that writing `props.fieldName` isn't all that long, and in a complex component preferable because then you know that the value comes directly from the props (and not from a local variable).

Reply:
The issue I've found is with the useEffect, if you have a lot of props.<prop> in the dependency array react's eslint auto fix changes your dependency to simply be `props` instead of `props.*`

Reply:
TIL